src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -582,36 +582,78 @@ bool fx_muxer_t::resolve_sdk_dotnet_path(const pal::string_t& own_dir, pal::stri
         trace::verbose(_X("Failed to obtain current working dir"));
     }
 
-    pal::string_t retval;
-    if (!global.empty())
+    std::vector<pal::string_t> hive_dir;
+    pal::string_t env_lookup;
+
+    // Multi-level SDK lookup can be disabled by setting DOTNET_MULTILEVEL_LOOKUP env var to zero
+    bool multilevel_lookup = true;
```

Comment:
This check is present in https://github.com/dotnet/core-setup/blob/c0aac4d98d3af034d8b9f8ca54ed49ec766db320/src/corehost/cli/fxr/fx_muxer.cpp#L380 as well. Can you please refactor this out to a single helper function that can be used in both the locations?